src/main/java/org/helpberkeley/memberdata/Restaurant.java

```diff
@@ -47,14 +47,13 @@ protected Restaurant(String name, int lineNumber) {
     }
 
     public static Restaurant createRestaurant(ControlBlock controlBlock, String name, int lineNumber) {
-        switch (controlBlock.getVersion()) {
-            case Constants.CONTROL_BLOCK_VERSION_200:
-                return new RestaurantV200(controlBlock, name, lineNumber);
-            case Constants.CONTROL_BLOCK_VERSION_300:
-                return new RestaurantV300(controlBlock, name, lineNumber);
-            default:
-                throw new MemberDataException("Control block version " + controlBlock.getVersion()
-                        + " is not supported for restaurant creation");
+        if (controlBlock.versionIsCompatible(Constants.CONTROL_BLOCK_VERSION_200)) {
+            return new RestaurantV200(controlBlock, name, lineNumber);
+        } else if (controlBlock.versionIsCompatible(Constants.CONTROL_BLOCK_VERSION_300)) {
+            return new RestaurantV300(controlBlock, name, lineNumber);
+        } else {
+            throw new MemberDataException("Control block version " + controlBlock.getVersion()
+                    + " is not supported for restaurant creation");
```

Comment:
Create a public constant in the file and for use with MessageFormat.format. Update test(s) to use MessageFormat.format and remove string duplication.